File: app2/src/main/java/com/ssz/framejava/base/ui/dagger/di/module/AppModule.java
package com.ssz.framejava.base.ui.dagger.di.module;


import com.ssz.framejava.app.AppContext;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

/**
 * @author : zsp
 * time : 2019 11 2019/11/19 12:45
 */
@Module
public class AppModule {

    AppContext context;

    public AppModule(AppContext context) {
        this.context = context;
    }

    @Singleton
    @Provides
    public AppContext provideAppContext() {
        return context;
    }
}

File: app/src/main/java/com/ssz/studydemo/base/ui/dagger/di/module/AppModule.kt
package com.ssz.studydemo.base.ui.dagger.di.module

import com.ssz.studydemo.app.AppContext
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

/**
 * @author : zsp
 * time : 2019 12 2019/12/11 14:28
 */
@Module
class AppModule(private val appContext: AppContext) {

    @Singleton
    @Provides
    fun provideAppContext(): AppContext {
        return appContext
    }
}